snapshots/helpers_test.go

```diff
@@ -110,28 +123,29 @@ func (m *mockSnapshotter) Restore(
 		return snapshottypes.SnapshotItem{}, errors.New("already has contents")
 	}
 
+	var item snapshottypes.SnapshotItem
 	m.items = [][]byte{}
 	for {
-		item := &snapshottypes.SnapshotItem{}
-		err := protoReader.ReadMsg(item)
+		item.Reset()
```

Comment:
is there a test case to test this in handled correctly? I recently ran into an issue where we used .Reset like this and the api is different than what we expected

Reply:
```
func (m *SnapshotItem) Reset()         { *m = SnapshotItem{} }
```

looks straightforward enough? what issue did you have? Do you concern that after `ReadMsg`, some fields still has the values from the last one?